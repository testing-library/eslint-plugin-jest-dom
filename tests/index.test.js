const { rules, generateRecommendedConfig, configs } = require("../lib/index");

it("should have all the rules", () => {
  expect(rules).toMatchSnapshot();
});

it("should have a recommended config with recommended rules", () => {
  expect(
    generateRecommendedConfig({
      good: { meta: { docs: { recommended: true } } },
      bad: { meta: { docs: { recommended: false } } }
    })
  ).toEqual({ "jest-dom/good": "error" });
});
