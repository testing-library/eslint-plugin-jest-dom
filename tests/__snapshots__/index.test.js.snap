// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have all the rules 1`] = `
Object {
  "prefer-checked": Object {
    "create": [Function],
    "meta": Object {
      "docs": Object {
        "category": "jest-dom",
        "description": "prefer toBeChecked over checking attributes",
        "recommended": true,
        "url": "prefer-checked",
      },
      "fixable": "code",
    },
  },
  "prefer-enabled-disabled": Object {
    "create": [Function],
    "meta": Object {
      "docs": Object {
        "category": "jest-dom",
        "description": "prefer toBeDisabled or toBeEnabled over checking attributes",
        "recommended": true,
        "url": "prefer-enabled-disabled",
      },
      "fixable": "code",
    },
  },
  "prefer-required": Object {
    "create": [Function],
    "meta": Object {
      "docs": Object {
        "category": "jest-dom",
        "description": "prefer toBeRequired over checking properties",
        "recommended": true,
        "url": "prefer-required",
      },
      "fixable": "code",
    },
  },
  "prefer-to-have-attribute": Object {
    "create": [Function],
    "meta": Object {
      "docs": Object {
        "description": "prefer toHaveAttribute over checking  getAttribute/hasAttribute ",
        "recommended": false,
      },
      "fixable": "code",
    },
  },
}
`;
